sky/server/common.py

```diff
@@ -128,24 +129,55 @@ class ApiServerInfo:
     commit: Optional[str] = None
 
 
-def get_api_cookie_jar_path() -> str:
-    return os.environ.get(server_constants.API_COOKIE_FILE_ENV_VAR,
-                          server_constants.API_COOKIE_FILE_DEFAULT_LOCATION)
+def get_api_cookie_jar_path() -> pathlib.Path:
+    """Returns the Path to the API cookie jar file."""
+    return pathlib.Path(
+        os.environ.get(server_constants.API_COOKIE_FILE_ENV_VAR,
+                       server_constants.API_COOKIE_FILE_DEFAULT_LOCATION)
+    ).expanduser().resolve()
 
 
 def get_api_cookie_jar() -> requests.cookies.RequestsCookieJar:
     """Returns the cookie jar used by the client to access the API server."""
     cookie_jar = requests.cookies.RequestsCookieJar()
-    cookie_file = get_api_cookie_jar_path()
-    if cookie_file:
-        cookie_path = pathlib.Path(cookie_file).expanduser().resolve()
-        if cookie_path.exists():
-            file_cookie_jar = MozillaCookieJar(cookie_path)
-            file_cookie_jar.load()
-            cookie_jar.update(file_cookie_jar)
+    cookie_path = get_api_cookie_jar_path()
+    if cookie_path.exists():
+        file_cookie_jar = MozillaCookieJar(cookie_path)
+        file_cookie_jar.load()
+        cookie_jar.update(file_cookie_jar)
     return cookie_jar
 
 
+def get_cookies_from_response(
+        response: 'requests.Response') -> requests.cookies.RequestsCookieJar:
+    """Returns the cookies from the API server response."""
+    server_url = get_server_url()
+    cookies = response.cookies
+    for prev_resp in response.history:
+        for cookie in prev_resp.cookies:
+            if cookie.domain in server_url:
+                cookies.set_cookie(cookie)
+    return cookies
+
+
+def set_api_cookie_jar(cookie_jar: CookieJar,
+                       create_if_not_exists: bool = True) -> None:
+    """Updates the file cookie jar with the given cookie jar."""
+    cookie_path = get_api_cookie_jar_path()
+    if not cookie_path.exists() and create_if_not_exists:
+        cookie_path.parent.mkdir(parents=True, exist_ok=True)
+        cookie_path.touch()
+    elif not cookie_path.exists() and not create_if_not_exists:
+        # if the file doesn't exist and we don't want to create it, do nothing
+        return
+
+    file_cookie_jar = MozillaCookieJar(cookie_path)
+    file_cookie_jar.load()
+    for cookie in cookie_jar:
+        file_cookie_jar.set_cookie(cookie)
+    file_cookie_jar.save()
```

Comment:
Suggested fix:
```suggestion
    if not cookie_path.exists() and not create_if_not_exists:
        # if the file doesn't exist and we don't want to create it, do nothing
        return

    file_cookie_jar = MozillaCookieJar(cookie_path)
    if cookie_path.exists():
      file_cookie_jar.load()
    for cookie in cookie_jar:
        file_cookie_jar.set_cookie(cookie)
    file_cookie_jar.save()
```
`file_cookie_jar.save()` will create the file if it doesn't already exist

Reply:
alright, added this